src/engine/source/builder/builders/opBuilderHelperMap.cpp

```diff
@@ -1412,6 +1415,91 @@ base::Expression opBuilderHelperEpochTimeFromSystem(const std::string& targetFie
         });
 }
 
+// field: +date_from_epoch/<$epoch_field_ref>|epoch_field
+base::Expression opBuilderHelperDateFromEpochTime(const std::string& targetField,
+                                                  const std::string& rawName,
+                                                  const std::vector<std::string>& rawParameters,
+                                                  std::shared_ptr<defs::IDefinitions> definitions,
+                                                  std::shared_ptr<schemf::ISchema> schema)
+{
+    auto parameters = helper::base::processParameters(rawName, rawParameters, definitions);
+
+    // Check parameters
+    helper::base::checkParametersSize(rawName, parameters, 1);
+
+    const auto name = helper::base::formatHelperName(rawName, targetField, parameters);
+
+    // If target field (key) is a schema field, the value should be a string
+    if (schema->hasField(DotPath::fromJsonPath(targetField))
+        && (schema->getType(DotPath::fromJsonPath(targetField)) != json::Json::Type::String))
+    {
+        throw std::runtime_error(
+            fmt::format("Engine helper builder: [{}] failed schema validation: Target field '{}' value is not a string",
+                        name,
+                        targetField));
+    }
+
+    // Tracing
+    const std::string successTrace {fmt::format(TRACE_SUCCESS, name)};
+    const std::string failureTrace1 {fmt::format(TRACE_REFERENCE_NOT_FOUND, name, parameters[0].m_value)};
+    const std::string failureTrace2 {fmt::format(TRACE_REFERENCE_TYPE_IS_NOT, "number", name, parameters[0].m_value)};
+    const std::string failureTrace3 {fmt::format("[{}] -> Failure: Value overflow", name)};
+    const std::string failureTrace4 {fmt::format("[{}] -> Failure: Couldn't create int from parameter", name)};
+
+    // Return result
+    return base::Term<base::EngineOp>::create(
+        name,
+        [=, targetField = std::move(targetField), parameter = std::move(parameters[0])](
+            base::Event event) -> base::result::Result<base::Event>
+        {
+            int IntResolvedParameter;
+            // Check parameter
+            if (helper::base::Parameter::Type::REFERENCE == parameter.m_type)
+            {
+                const auto paramValue = event->getInt(parameter.m_value);
+                if (!paramValue.has_value())
+                {
+                    return base::result::makeFailure(
+                        event, (!event->exists(parameter.m_value) ? failureTrace1 : failureTrace2));
+                }
```

Comment:
If getInt returns empty, check the Int64. I would even check this first since the epoc variable is 64 bit.